src/main/java/org/opensearch/flowframework/workflow/AbstractRetryableWorkflowStep.java

```diff
@@ -110,25 +91,36 @@ void retryableGetMlTask(
                 try {
                     logger.info(workflowStep + " successful for {} and modelId {}", workflowId, response.getModelId());
                     String resourceName = WorkflowResources.getResourceByWorkflowStep(getName());
+                    String id;
                     if (getName().equals(WorkflowResources.DEPLOY_MODEL.getWorkflowStep())) {
-                        completeFuture(resourceName, nodeId, workflowId, response, future);
+                        id = response.getModelId();
                     } else {
-                        flowFrameworkIndicesHandler.updateResourceInStateIndex(
-                            workflowId,
-                            nodeId,
-                            getName(),
-                            response.getTaskId(),
-                            ActionListener.wrap(updateResponse -> {
-                                logger.info("successfully updated resources created in state index: {}", updateResponse.getIndex());
-                                completeFuture(resourceName, nodeId, workflowId, response, future);
-                            }, exception -> {
-                                logger.error("Failed to update new created resource", exception);
-                                future.completeExceptionally(
-                                    new FlowFrameworkException(exception.getMessage(), ExceptionsHelper.status(exception))
-                                );
-                            })
-                        );
+                        id = response.getTaskId();
                     }
+                    flowFrameworkIndicesHandler.updateResourceInStateIndex(
+                        workflowId,
+                        nodeId,
+                        getName(),
+                        id,
+                        ActionListener.wrap(updateResponse -> {
+                            logger.info("successfully updated resources created in state index: {}", updateResponse.getIndex());
+                            future.complete(
+                                new WorkflowData(
+                                    Map.ofEntries(
+                                        Map.entry(resourceName, response.getModelId()),
+                                        Map.entry(REGISTER_MODEL_STATUS, response.getState().name())
```

Comment:
Not sure we need "register model status" for a deployed model, but doesn't hurt to have extra field in the output.

Reply:
here we are updating just the workflowData and not the index, so just thought on the same line that it couldn't hurt to add this.